docs/pattern-matching.md

```diff
@@ -88,28 +88,37 @@ To ensure that recursive pattern matching functions don't loop in strict mode, i
 ```py
 # This is what the Foo function actually compiles to.
 # With -Olinearize-matches and -Ofloat-combinators (default on strict mode)
-(Foo) = λa λb λc (switch a { 0: Foo$C5; _: Foo$C8 } b c)
-
-(Foo$C5) = λd λe (d Foo$C0 Foo$C4 e) # Foo.case_0
-(Foo$C0) = λ* B                      # Foo.case_0.case_true
-(Foo$C4) = λg (g Foo$C3 B)           # Foo.case_0.case_false
-(Foo$C3) = λh λi (i Foo$C1 Foo$C2 h) # Foo.case_0.case_false.case_cons
-(Foo$C1) = λj λk λl (A l j k)        # Foo.case_0.case_false.case_cons.case_cons
-(Foo$C2) = λ* B                      # Foo.case_0.case_false.case_cons.case_nil
-
-(Foo$C8) = λn λo λ* (o Foo$C6 Foo$C7 n) # Foo.case_+
-(Foo$C6) = λ* 0                         # Foo.case_+.case_true
-(Foo$C7) = λr (+ r 1)                   # Foo.case_+.case_false
+
+unchecked Foo__C0: _
```

Comment:
Remove the type annotations here since they're only noise.

We should probably also add a note here that as a user you can't write a function with `__` in its name, that sequence is reserved for things generated by the compiler.